### PR TITLE
updating reference tests for protectionsState

### PR DESCRIPTION
**Pull request by shakyShane**
Task/Issue URL: https://app.asana.com/0/0/1205924265528360/f

### Description

Updating tests to suit this recent change https://github.com/duckduckgo/privacy-reference-tests/pull/112

### Steps to test this PR

- [x] when latest reference tests run, they will pass with this PR

### UI changes
| Before  | After |
| ------ | ----- |
!(Upload before screenshot)|(Upload after screenshot)|

### CODE REVIEW

**Comment by shakyShane** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3821** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3821" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3821?utm_source=stack-comment).

File: app/src/test/java/com/duckduckgo/app/referencetests/brokensites/BrokenSitesReferenceTest.kt [hunk 1]
```diff
@@ -134,6 +138,11 @@ class BrokenSitesReferenceTest(private val testCase: TestCase) {
             PrivacyConfigData(version = testCase.remoteConfigVersion ?: "v", eTag = testCase.remoteConfigEtag ?: "e"),
         )
 
+        if (!testCase.protectionsEnabled) {
```
Comment (marcosholgado) [inline app/src/test/java/com/duckduckgo/app/referencetests/brokensites/BrokenSitesReferenceTest.kt:141]: Can simplify to:

```
val url = Uri.parse(testCase.siteURL).host
whenever(mockUserAllowListRepository.isDomainInUserAllowList(url)).thenReturn(!testCase.protectionsEnabled)
```

**Review by marcosholgado — CHANGES_REQUESTED** [review]
Small change to simplify a bit.

**Review by marcosholgado — APPROVED** [review]
LGTM!

**Pull request merged**